src/middleware/basetypes.rs

```diff
@@ -125,6 +125,32 @@ impl fmt::Display for RawValue {
     }
 }
 
+impl ToHex for RawValue {
```

Comment:
One question, I see that before this PR, the type `Hash` had implementations for the traits `ToHex` and `FromHex`. Here it is introducing `ToHex` for the type `RawValue`; should we add also the implementation of `FromHex` for the type `RawValue`? (so that it is available in the future for when converting back and forth the `RawValue` from & to hex)

Reply:
I've added the `FromHex` trait as well.